src/core/test/spec.js

```diff
@@ -104,64 +105,89 @@ describe('Test Passables "test" function', () => {
+                        test(f5, lorem.sentence(), async() => await rejectLater());
                     });
                 });
 
                 it('Should fail for rejected promise', (done) => {
-                    expect(output.hasErrors(f1)).to.equal(false);
-                    expect(output.hasErrors(f2)).to.equal(false);
-                    expect(output.hasErrors(f4)).to.equal(false);
+                    [f1, f2, f3, f5].forEach((field) =>
+                        expect(output.hasErrors(field)).to.equal(false)
+                    );
+
+                    setTimeout(() => {
+                        [f1, f2, f3, f5].forEach((field) =>
+                            expect(output.hasErrors(field)).to.equal(true)
+                        );
+
+                        done();
+                    }, 550);
+                });
+
+                it('Should fail with rejection message when provided', (done) => {
                     setTimeout(() => {
-                        expect(output.hasErrors(f1)).to.equal(true);
-                        expect(output.hasErrors(f2)).to.equal(true);
-                        expect(output.hasErrors(f4)).to.equal(true);
+                        expect(output.getErrors(f3)).to.deep.equal([rejectionMessage]);
+
                         done();
                     }, 550);
                 });
 
                 it('Should pass for fulfilled promises', (done) => {
-                    expect(output.hasErrors(f3)).to.equal(false);
+                    expect(output.hasErrors(f4)).to.equal(false);
+
                     setTimeout(() => {
-                        expect(output.hasErrors(f3)).to.equal(false);
+                        expect(output.hasErrors(f4)).to.equal(false);
                         done();
                     }, 500);
                 });
             });
 
             describe('When passing a Promise as a test', () => {
                 describe('failing', () => {
-                    let f1, f2;
+                    let f1, f2, f3;
+                    const rejectionMessage = lorem.sentence();
+
                     beforeEach(() => {
                         f1 = lorem.word();
                         f2 = lorem.word();
+                        f3 = lorem.word();
 
-                        output = passable(lorem.word(), (test, draft) => {
-                            test(f1, lorem.sentence(), new Promise((resolve, reject) => setImmediate(reject)));
-                            test(f2, lorem.sentence(), new Promise((resolve) => setTimeout(resolve)));
-                            test(f2, lorem.sentence(), new Promise((resolve) => setTimeout(resolve, 500)));
+                        output = passable(lorem.word(), (test) => {
+                            test(f1, lorem.sentence(), new Promise((_, reject) => setImmediate(reject)));
+                            test(f2, new Promise((_, reject) => setImmediate(() => reject(rejectionMessage))));
```

Comment:
At the moment you only added tests with either rejectionMessage or statement, not both. We don't test the case in which the rejectionMessage takes precedence (according to your current implementation).